### ISSUE TITLE
GoogleCloudEnterpriseSearchRetriever returned 'datastore not found' error even with the 'us' configurations

### ISSUE THREAD

**Issue by mhyeonsoo** [langchain-ai/langchain#15785]
### System Info

langchain==0.0.352

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

```python
from langchain.retrievers import (
    GoogleVertexAIMultiTurnSearchRetriever,
    GoogleVertexAISearchRetriever,
    GoogleCloudEnterpriseSearchRetriever
)
    
PROJECT_ID = "my_project_id"
SEARCH_ENGINE_ID = "I tried both for datastore_id and app_id at Vertex Search"
LOCATION_ID = "us"

retriever = GoogleCloudEnterpriseSearchRetriever(
    project_id=PROJECT_ID,
    search_engine_id=SEARCH_ENGINE_ID,
    location_id=LOCATION_ID,
    max_documents=3
)

while 1:
    message = input()
    result = retriever.get_relevant_documents(message)
    for doc in result:
        print(doc)
```

### Expected behavior

I expected it works well with the defined datastore, but it returned the error saying,

```
google.api_core.exceptions.NotFound: 404 DataStore projects/500618827687/locations/us/collections/default_collection/dataStores/['datastore_id'] not found
```

**Comment by mhyeonsoo** [comment]
Thanks @dosu-bot,

I upgraded langhchain to ```(v0.1.0)```, and also changed search engine location from ```us``` to ```global``` just for the case.

For now, I am getting different error which is,
```
raise _InactiveRpcError(state)  # pytype: disable=not-instantiable
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "Stream removed"
	debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"Stream removed", grpc_status:2, created_time:"2024-01-10T09:25:49.088826+09:00"}"
```
and it ends up with
```
in error_remapped_callable
    raise exceptions.from_grpc_error(exc) from exc
google.api_core.exceptions.Unknown: None Stream removed
```

**Comment by mhyeonsoo** [comment]
@dosu-bot

this is my grpcio version.
```1.60.0```

and with the sleep(1) method, I am still getting an error.

**Comment by Adi8885** [comment]
Hi @mhyeonsoo ,

I was facing the same issue which was addressed in a previous issue https://github.com/langchain-ai/langchain/issues/10580 and was fixed in version [v0.0.310](https://github.com/langchain-ai/langchain/releases/tag/v0.0.310) by @holtskinner 

I'm able to mitigate this issue with following setup :
Python : 3.10
Langchain : 0.1.1
Discovery Engine Version : 0.10.0
```
pip install google-cloud-discoveryengine==0.10.0 langchain==0.1.1
```

Let me know if this helps

**Comment by mhyeonsoo** [comment]
Thanks @Adi8885 ,

For my case, I successfully accessed to my data store with the latest version of langchain, and by changing the importing to 
```python
From langchain_community.retrievers
```

Thanks for your advice!
